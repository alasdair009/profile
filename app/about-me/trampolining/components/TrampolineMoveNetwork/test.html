<html lang="en">
  <head>
    <style>
      .videos {
        display: flex;
        height: 100vh;
        width: 100vw;
      }
      .videos div {
        align-items: center;
        display: flex;
        flex-direction: column;
        justify-content: center;
        width: 50%;
      }
      iframe {
        aspect-ratio: 16/9;
        width: 90%;
      }
    </style>
  </head>
  <body>
    <div class="videos">
      <div>
        <iframe
          id="standard"
          src=""
          data-src="https://www.youtube.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Standard Embed"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goStandard">Standard</button>
      </div>
      <div>
        <iframe
          id="privacy"
          src=""
          data-src="https://www.youtube-nocookie.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Enhanced privacy"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goPrivacy">Privacy</button>
      </div>
    </div>
    <script>
      const goStandard = document.getElementById("goStandard");
      const standardIFrame = document.getElementById("standard");
      const goPrivacy = document.getElementById("goPrivacy");
      const privacyIFrame = document.getElementById("privacy");
      goStandard.addEventListener(
        "click",
        function () {
          standardIFrame.setAttribute("src", standardIFrame.getAttribute("data-src"));
        },
        false
      );
      goPrivacy.addEventListener(
        "click",
        function () {
          privacyIFrame.setAttribute("src", privacyIFrame.getAttribute("data-src"));
        },
        false
      );
    </script>
  </body>
</html>
