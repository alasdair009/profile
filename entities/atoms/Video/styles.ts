"use client";
import styled from "styled-components";

export const Root = styled.video`
  object-fit: cover;
`;
