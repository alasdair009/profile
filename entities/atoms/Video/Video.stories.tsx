import { Video } from "./Video";
import { Meta, StoryObj } from "@storybook/react";
import trampolinePosterImage from "../../assets/trampoline-poster.webp";
import { expect, within } from "@storybook/test";

const meta: Meta<typeof Video> = {
  component: Video,
  args: {
    poster: trampolinePosterImage,
    webmSrc:
      "https://ftjjvjgljna2ohz8.public.blob.vercel-storage.com/trampoline-cambs.webm",
  },
};
export default meta;

export const Default: StoryObj<typeof Video> = {
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const videoElement = canvas.getByTestId(Video.name);

    await expect(videoElement).toBeInTheDocument();
    await expect(videoElement).toHaveAttribute(
      "poster",
      trampolinePosterImage.src
    );
    await expect(videoElement.children[0]).toHaveAttribute("src", args.webmSrc);
  },
};
