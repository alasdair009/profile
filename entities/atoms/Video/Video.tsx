"use client";
import { Root } from "./styles";
import { HTMLAttributes } from "react";
import { StaticImageData } from "next/image";
import { prefersReducedMotion } from "@/entities";

type VideoProps = {
  /**
   * Repeat the video after it finishes.
   */
  loop?: boolean;
  /**
   * Autoplay on load (note this has browser limitations).
   */
  autoPlay?: boolean;
  /**
   * Mute the video.
   */
  muted?: boolean;
  /**
   * Url or path to the mp4 file (needed for iOS).
   */
  mp4Src?: string;
  /**
   * Url or path to the webm file.
   */
  webmSrc: string;
  /**
   * Graphic to display when the video is not available.
   */
  poster: StaticImageData;
} & HTMLAttributes<HTMLVideoElement>;

/**
 * Play a video within the interface.
 */
export function Video({
  mp4Src,
  webmSrc,
  loop = true,
  autoPlay = true,
  muted = true,
  poster,
  ...rest
}: VideoProps) {
  return (
    <Root
      loop={loop}
      autoPlay={!prefersReducedMotion}
      playsInline={autoPlay}
      muted={muted}
      poster={poster.src}
      data-testid={Video.name}
      {...rest}
    >
      <source type="video/webm" src={webmSrc} />
      {mp4Src && <source type="video/mp4" src={mp4Src} />}
    </Root>
  );
}
