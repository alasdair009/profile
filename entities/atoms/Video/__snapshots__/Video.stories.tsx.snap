// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/Video Default play-test 1`] = `
<video loop
       autoplay
       playsinline
       poster="static/media/entities/assets/trampoline-poster.webp"
       data-testid="Video"
       class="sc-braxZu jSTinX"
>
  <source type="video/webm"
          src="https://ftjjvjgljna2ohz8.public.blob.vercel-storage.com/trampoline-cambs.webm"
  >
</video>
`;
